app/views/shared/_contact_panels.html.erb

```diff
@@ -113,111 +114,173 @@
+        var sWOChatElement = document.createElement('a');
+        var sWOChatElementText = document.createTextNode("<%= t('contact_panels.chat.available.call_to_action') %>");
+        sWOChatElement.className = "contact-panel__button button t-chat-button";
+        sWOChatElement.appendChild(sWOChatElementText);
 
 
-            chatCTA.innerHTML = "";
-            chatCTA.appendChild(sWOChatElement);
-            chatDescription.innerHTML = "<%= t('contact_panels.chat.available.description') %>";
-            chatContent.className = "contact-panel__dynamic-content is-loaded";
+        chatCTA.innerHTML = "";
+        chatCTA.appendChild(sWOChatElement);
+        chatDescription.innerHTML = "<%= t('contact_panels.chat.available.description') %>";
+        chatContent.className = "contact-panel__dynamic-content is-loaded";
 
 
         <% if translation?('contact_panels.chat.smallprint') %>
-            chatDescription.innerHTML += '&nbsp;*';
+          chatDescription.innerHTML += '&nbsp;*'; 
         <% end %>
 
 
-            sWOChatElement.onclick = function() {
-                window.open(sWOChatElement.href, "Chat", "width=484,height=361,scrollbars=yes,resizable=yes");
-                return false;
-            };
+          sWOChatElement.onclick = function () {
+            window.open(sWOChatElement.href, "Chat", "width=484,height=361,scrollbars=yes,resizable=yes");
+            return false;
+          };
 
-            sWOChatElement.href = sWOChatstart;
-            sWOChatElement.target = "_blank";
+        sWOChatElement.href = sWOChatstart;
+        sWOChatElement.target = "_blank";
 
         <% if show_floating_chat? %>
-            chatPopup.className = "chat-box__container";
-            chatPopup.href = sWOChatstart;
-            chatPopup.target = "_blank";
-
-            chatPopup.onclick = function() {
-                window.open(chatPopup.href, "Chat", "width=484,height=361,scrollbars=yes,resizable=yes");
-                return false;
-            };
+          chatPopup.className = "chat-box__container";
+          chatPopup.href = sWOChatstart;
+          chatPopup.target = "_blank";
+
+          chatPopup.onclick = function () {
+            window.open(chatPopup.href, "Chat", "width=484,height=361,scrollbars=yes,resizable=yes");
+            return false;
+          }; 
         <% end %>
 
-        }
+      }
 
     <% end %>
 
-    };
-
-    if (typeof sWOTrackPage == 'function') {
-        sWOTrackPage();
+  };
 
-    <%# Track page enagagement (a visit lasting more than 30 seconds) %>
-        setTimeout("sWOInvite='';sWOResponse='N';sWOPage=(sWOPage || '')+'%3Ftimer%3Dtrue';sWOTrackPage();", 30000);
-    }
+  if (typeof sWOTrackPage == 'function') {
+    sWOTrackPage();
 
+  <% #Track page enagagement(a visit lasting more than 30 seconds) %>
+      setTimeout("sWOInvite='';sWOResponse='N';sWOPage=(sWOPage || '')+'%3Ftimer%3Dtrue';sWOTrackPage();", 30000);
+  }
 </script>
 <% end %>
 <% end %>
```

Comment:
This file is a candidate for refactoring. For example, what is the difference between `if chat_opening_hours.open?` and `if chat_opening_hours.status_as_string == 'open'`? Are there opportunities to partialize any of the content?

Reply:
@margOnline just noticed that all of these instances of `chat_opening_hours.status_as_string == 'open'` are on the WhatsApp contact panel which is going to be reworked on [**TP 10511**](https://moneyadviceservice.tpondemand.com/RestUI/Board.aspx#page=board/5682182231901749200&appConfig=eyJhY2lkIjoiNTM0ODcyOThDQjBDMTMyMDgyMTBENEZEM0QwM0ZBQzkifQ==&boardPopup=userstory/10511/silent) so we can address these issues on that ticket and make the content consistent as well.

Reply:
@margOnline All the JS in here is third party and I'd be reluctant to do much to it right now. It would be nice to pull it out as a discrete component and refactor it but I think that is something to be done in the future.

Reply:
@davidtrussler my comment was with reference to the markup and erb, not the js. I guess I should have positioned the comment at the beginning of the file 

@andreferreiradlw I kind of think that this would be an appropriate time to at least address the inconsistent erb code.

Reply:
@margOnline I've corrected the inconsistencies now regarding those instances of `chat_opening_hours.status_as_string == 'open'`